ktor-server/ktor-server-plugins/ktor-server-rate-limit/jvmAndNix/src/io/ktor/server/plugins/ratelimit/Routing.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2014-2022 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.plugins.ratelimit
+
+import io.ktor.server.application.*
+import io.ktor.server.routing.*
+import io.ktor.util.*
+
+/**
+ * Creates a route with Rate-Limit rules applied to it.
+ * This function accepts name of RateLimit providers defined in the [RateLimit] plugin configuration.
+ * @see [RateLimit]
+ *
+ * @param configuration names of RateLimit providers defined in the [RateLimit] plugin configuration.
+ * @throws IllegalArgumentException if there are no registered providers referred by [configuration] names.
+ */
+public fun Route.rateLimit(
+    configuration: String = LIMITER_NAME_EMPTY,
```

Comment:
offtop: may be we should try to be more type safe here and in auth (may be in other places) and introduce some types keys like ktor `Attributes` / coroutines `CoroutinesContext`
So it will be harder to mismatch - strings are really easy to misuse.

Reply:
The main benefit of `AttributesKey` and `CoroutinesContextElement` is generics, to provide type safety.
We can add value class here, but I'm not sure how helpful it would be here.

Reply:
yeah, while I was thinking just about interface in library and object as key on consumer side, value class could be better.
And I think, that this is exactly one of the good use cases for value(inline) classes.

> but I'm not sure how helpful it would be here.

as for now, if library consumer will use this `configuration` to link plugin and routing declaration she will have to store somewhere this `String` key, and in declaration only it name will tell readers for what this key is used, like
`val rateLimiterForCustomersApi = "name"`
When using interface/object of value class, declaration will be much more understandable for reader, like:
`val rateLimiterForCustomersApi = RateLimiter.Config("name")`
or
`object RateLimiterForCustomersApi: RateLimiter.Config`
(names are of course TBD)

Reply:
I made it value class